### PR TITLE
Documentation & refactoring for custom kubectl from source

### PR DESCRIPTION
**Pull request by smukherj1**
Addresses #245

### CODE REVIEW

File: toolchains/kubectl/kubectl_configure.bzl [hunk 1]
```diff
@@ -74,21 +109,33 @@ def kubectl_configure(name, **kwargs):
...
-            sha256 = "21160531ea8a9a4001610223ad815622bf60671d308988c7057168a495a7e2e8",
-            strip_prefix = "repo-infra-b4bc4f1552c7fc1d4654753ca9b0e5e13883429f",
-            urls = ["https://github.com/kubernetes/repo-infra/archive/b4bc4f1552c7fc1d4654753ca9b0e5e13883429f.tar.gz"],
+            sha256 = _k8s_repo_tools_sha,
+            strip_prefix = "{}-{}".format(
+                _k8s_repo_tools_repo,
+                _k8s_repo_tools_commit
+            ),
+            urls = ["https://github.com/{}/{}/archive/{}.tar.gz".format(
+                _k8s_org,
```
Comment (nlopezgi) [inline toolchains/kubectl/kubectl_configure.bzl]: lets enable overriding these values as well

**Comment by smukherj1** [comment]
@nlopezgi Unfortunately, with v1.10.0 I get 
```
E1218 16:38:19.395878       2 main.go:51] Error: Failed making a parser: unable to add directory "./vendor/k8s.io/apimachinery/pkg/util/sets/types": unable to import "./vendor/k8s.io/apimachinery/pkg/util/sets/types": cannot find package "./vendor/k8s.io/apimachinery/pkg/util/sets/types" in:
        /usr/local/google/home/suvanjan/.cache/bazel/_bazel_suvanjan/a1bcb3b63a242c66f0c6010821551247/sandbox/linux-sandbox/31/execroot/io_bazel_rules_k8s/vendor/k8s.io/apimachinery/pkg/util/sets/types
```

Seems to be because https://github.com/kubernetes/kubernetes/blob/fc32d2f3698e36b93322a3465f63a14e9f0eaead/vendor/k8s.io/apimachinery points to an invalid target?

Verified it works with v1.13.1 though. Addressed other comments. ptal

File: toolchains/kubectl/README.md [hunk 2]
```diff
@@ -8,23 +8,74 @@ rule. At the moment, the tool's configuration does one of the following:
...
+    k8s_repo_tools_commit = "b4bc4f1552c7fc1d4654753ca9b0e5e13883429f",
+    k8s_repo_tools_sha256 = "21160531ea8a9a4001610223ad815622bf60671d308988c7057168a495a7e2e8",
+    k8s_repo_tools_prefix = "repo-infra-b4bc4f1552c7fc1d4654753ca9b0e5e13883429f"
+)
+```
+
+Note that by default the `k8s_repositories()` calls `kubectl_configure` if it
+hasn't already been called. This configures a `kubectl_toolchain` that looks for
+the `kubectl` tool in the sytem path. If no `kubectl` tool is found, trying to
+execute `bazel run` for any targets will not work.
```
Comment (nlopezgi) [inline toolchains/kubectl/README.md]: Also add note user must find out the right version of repository infrastructure tools that works with the version of kubectl sources they want to use by looking at the io_kubernetes_build target in https://github.com/kubernetes/kubernetes/blob/{k8s_commit}/build/root/WORKSPACE

**Comment by smukherj1** [comment]
@nlopezgi Doc added. ptal

**Pull request merged**